pkg/importer/spanner/sysl.arrai

```diff
@@ -0,0 +1,140 @@
+    cond {
+        hasAttributePattern(entity, attr): token
+    };
+
+# entityPatternDelimiter returns the token based on entity value
+let entityPatternDelimiter = \entity \model \token
+    cond {
+        hasEntityPattern(entity, model): token
+    };
+
+# sortingOrder determines and appends sorting order
+let sortingOrder = \e \attr
+    let keyOrder = //seq.split('(', //seq.join('', (e.primary_key where //seq.contains(attr.name, .)) orderby .));
+    cond {
+        keyOrder count > 1: cond keyOrder(1) {
+            'asc)': ', ~asc',
+            'desc)': ', ~desc'
+        }
+    };
+
+# compareColumnOrder compares the col order between primary key and table
+let compareColumnOrder = \entity
+    let pk = (entity.primary_key orderby .) >> //seq.split('(', .) >> .(0);
+    cond {
+        pk & (entity.attributes >> .name): false,
+        _: true,
+    };
+
+# attributePattern generates the pattern constituents depending on the mode
+let attributePattern = \entity \attr \mode
+    cond {
+        mode = 0 && (entity.primary_key where //seq.contains(attr.name, .)): $`~pk${sortingOrder(entity, attr)}`,
+        let fks = (entity.foreign_keys => ((.foreign_keys where .attribute = attr.name) => .reference_attribute orderby .)) where .;
+        mode = 1 && fks: '~fk',
+        mode = 2 && attr.options:
+            let opt = //seq.split('=',attr.options);
+            $`${opt(0)}='${opt(1)}'`,
+        mode = 3 && attr.length = 'MAX': '~max',
+        mode = 4 && //seq.has_prefix('0x', attr.length): '~hex',
+    };
+
+# entityPattern generates the pattern constituents depending on the mode
+let entityPattern = \entity \model \mode
+    cond {
+        mode = 0 && entity.primary_key count > 1 && compareColumnOrder(entity):
+            $`primary_key="${//seq.join(', ', entity.primary_key orderby .)}"`,
+        mode = 1 && entity.cluster:
+            //seq.join('', entity.cluster >> $`interleave_in_parent="${.interleaved_in}", interleave_on_delete="${.on_delete}"`),
+        mode = 2 && entity.foreign_keys:
+            $`foreign_keys=[${//seq.join(', ',(entity.foreign_keys => \keys $`["constraint:${keys.constraint_name}` ++
+            $`","columns:${//seq.join(', ', (keys.foreign_keys => .attribute) orderby .)}"]`) orderby .)}]`,
+        let relatedIndices = ((model.indexes orderby .) where .@item.table_name = entity.name); mode = 3 && relatedIndices:
+            $`indexes=[${//seq.join(', ', ((model.indexes orderby .) where .@item.table_name = entity.name) >>
+            $`"name:${.table_name}` ++
+            $`","unique:${cond {(.unique = 'UNIQUE'):'true', _:'false'}}` ++
+            $`","null_filtered:${cond {(.nullfiltered = 'NULL_FILTERED'):'true', _:'false'}}` ++
+            $`","key_parts:${//seq.join(', ', .key_part orderby .)}` ++
+            cond {.storing_col: $`","storing:' ${//seq.join(', ',.storing_col)}`}  ++
+            cond {.interleaved_table: $`","interleave_in:' ${.interleaved_table}`})}"]`,
```

Comment:
You don't need to concat strings in a template. If you're worried about line length, try something that more cleanly maps to your intent:

```suggestion
            $`"${[
                $`name:${.table_name}`,
                $`unique:${cond {(.unique = 'UNIQUE'):'true', _:'false'}}`,
                $`null_filtered:${cond {(.nullfiltered = 'NULL_FILTERED'):'true', _:'false'}}`,
                $`key_parts:${.key_part orderby .::, }`,
                .storing_col && $`storing:' ${.storing_col::,}`,
                .interleaved_table && $`interleave_in:' ${.interleaved_table}`,
            ] where .@item::",":"}`
```
Not sure why those `'`s are there in `storing_col` and `interleaved_table`?

Also nit: are `.unique` and `.nullfiltered` ever anything other than `'UNIQUE'` and `'NULL_FILTERED'`? If not, just store a `true` or `false` instead and the `cond` will be simpler (or just use their truthiness like `.unique && 'true' || 'false'`).

Reply:
ok